### PR TITLE
Make auto_readahead_size default true

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: Make auto_readahead_size option default true

Test Plan: benchmarks and exisiting tests

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: db/db_iter.cc [hunk 1]
```diff
@@ -764,6 +765,7 @@ bool DBIter::ReverseToBackward() {
         saved_key_.GetUserKey(), kMaxSequenceNumber, kValueTypeForSeek));
     if (!expect_total_order_inner_iter() || auto_readahead_size_) {
       iter_.SeekForPrev(last_key.GetInternalKey());
+      // auto_readahead_size_ = false;
```
Comment (anand1976) [inline db/db_iter.cc:768]: Temp change to be removed?

File: file/prefetch_test.cc [hunk 2]
```diff
@@ -2353,314 +2353,6 @@ TEST_P(PrefetchTest1, SeekParallelizationTest) {
   Close();
 }
 
-// This test checks if readahead_size is trimmed when upper_bound is reached.
```
Comment (anand1976) [inline file/prefetch_test.cc:2356]: Are these tests not valid anymore?

Comment (akankshamahajan15) [reply file/prefetch_test.cc:2356]: No! These are for FindReadAheadSizeUpperBound that has been removed.

File: db/db_iter.cc (hunk 1 above)
Comment (akankshamahajan15) [reply db/db_iter.cc:768]: yes. I was trying to optimize backward scan.

File: file/prefetch_test.cc (hunk 2 above)
Comment (akankshamahajan15) [reply file/prefetch_test.cc:2356]: I've additional tests for sync scan for block cache lookup in async support PR.

**Review by anand1976 — APPROVED** [review]
LGTM

**Pull request closed without merging**